torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -362,14 +353,14 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupGloo::broadcast(
   }
 
   // Define how to run the algorithm and copy back results
-  entry->run = [tensors](EntryType& entry) mutable {
+  entry->run = [=]() mutable {
```

Comment:
Yep, it's a raw pointer though, so it's a cheap value.

Reply:
It is a heap pointer. The checkout function returns the pointer stored in the unique_ptr in the cache. Since we control when the cache will be emptied (for now only on destruction of the process group) we know it will remain valid at least until the worker thread has executed the related work. Passing a raw heap pointer around is cheaper than moving a unique_ptr around like I did in the first rev of this PR.

Reply:
Makes sense, thanks!